### PR TITLE
[dev-menu] update dev menu to match redesign

### PR DESCRIPTION
**Pull request by ajsmth**
# Why

This PR is the React changes made to the dev menu app. I've split this into a couple of commits - the first is the implementation of the React app, the second is all of the new icons that the designs use (these icon additions are mostly just noise in the git diffs)

In the next PR I'll include all of the native changes required for this menu - for now I've stubbed them out in JS

# How

This update primarily uses the `expo-dev-client-components` primitives to build out the new designs, and renders them in the repurposed bottom sheet. 

# Test Plan

Unit tests are a WIP and will be done in a separate PR

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-dev-menu/app/screens/MainScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    const { hostUrl } = buildInfo;
+    urlClipboard.onCopyPress({ hostUrl });
+  }
+
+  function onCopyBuildInfoPress() {
+    const { runtimeVersion, sdkVersion, appName, appVersion } = buildInfo;
+    buildInfoClipboard.onCopyPress({ runtimeVersion, sdkVersion, appName, appVersion });
+  }
+
+  const hasCopiedBuildInfoContext = Boolean(buildInfoClipboard.clipboardContent);
```
Comment (esamelson) [inline packages/expo-dev-menu/app/screens/MainScreen.tsx:44]: this will reset when the menu is closed and reopened, correct?

**Review by esamelson — APPROVED** [review]
just a few small comments/questions but mostly this looks great. thanks for splitting this up to make it faster to review! wishing I could have the exact thing we're building (that I could just scan a QR code and load this on my device right away) 😄

File: packages/expo-dev-menu/app/screens/MainScreen.tsx (hunk 1 above)
Comment (ajsmth) [reply packages/expo-dev-menu/app/screens/MainScreen.tsx:44]: yes - I think the "context" is a typo (meant to write "content") - its reset after ~3 seconds, and would be reset completely on mount / unmount

File: packages/expo-dev-menu/app/screens/MainScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+                </>
+              )}
+            </View>
+          </Row>
+
+          <Spacer.Horizontal size="flex" />
+
+          <Button.ScaleOnPressContainer bg="ghost" rounded="full" minScale={0.8}>
+            <View padding="micro">
+              <XIcon />
```
[the remark this answers is not in the document]
Comment (ajsmth) [reply packages/expo-dev-menu/app/screens/MainScreen.tsx:83]: 🙏 thank you for catching this!

File: packages/expo-dev-menu/app/screens/MainScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+            </View>
+          </Button.ScaleOnPressContainer>
+        </Row>
+      </View>
+
+      <Divider />
+
+      <View bg="default" padding="medium">
+        <Text color="secondary">Connected to local server</Text>
```
[the remark this answers is not in the document]
Comment (ajsmth) [reply packages/expo-dev-menu/app/screens/MainScreen.tsx:92]: this is a good point - will add some logic here in the mean time

**Review by lukmccall — APPROVED** [review]

File: packages/expo-dev-menu/app/screens/MainScreen.tsx [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+            <View padding="micro">
+              <XIcon />
+            </View>
+          </Button.ScaleOnPressContainer>
+        </Row>
+      </View>
+
+      <Divider />
+
+      {Boolean(buildInfo.hostUrl) && (
```
Comment (esamelson) [inline packages/expo-dev-menu/app/screens/MainScreen.tsx:103]: this is probably fine for now, but for the longer term i'm not sure this is the right logic here -- it seems like if we're viewing a published update we should be showing the URL to the update (manifest) here, whether that's `hostUrl` or a different field value

**Review by esamelson — APPROVED** [review]
looks great to me!

File: packages/expo-dev-menu/app/screens/MainScreen.tsx (hunk 4 above)
Comment (ajsmth) [reply packages/expo-dev-menu/app/screens/MainScreen.tsx:103]: yes I am meaning to follow up with you about getting the right value here!

Comment (esamelson) [reply packages/expo-dev-menu/app/screens/MainScreen.tsx:103]: I think the correct value would probably be DevLauncherController.manifestURL (or derived from it), may need to find a way to pass that value into here though.

**Pull request merged**